### PR TITLE
refactor(utility): use pa.array instead of np.array in LazyList

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2hd6fr1 直接使用 pa.array 替换 numpy.array](https://app.clickup.com/t/2hd6fr1)

File: graviti/utility/lazy.py [hunk 1]
```diff
@@ -215,7 +211,7 @@ def __init__(self, total_count: int, limit: int, getter: Callable[[int, int], An
         self._lists: List[LazyList[Any]] = []
 
     def create_list(
-        self, extractor: Callable[[Any], Iterable[Any]], dtype: "DTypeLike"
+        self, extractor: Callable[[Any], Iterable[Any]], dtype: pa.DataType
```
Comment (Lee-000) [inline graviti/utility/lazy.py:214]: The docstring needs to be updated.
Also, do we still need the dtype?

File: graviti/utility/lazy.py [hunk 2]
```diff
@@ -54,30 +50,30 @@ def __init__(
...
     def __getitem__(self, index: int) -> _T:
         ...
 
     @overload
-    def __getitem__(self, index: slice) -> "NDArray[_T]":
+    def __getitem__(self, index: slice) -> pa.list_:
         ...
 
-    def __getitem__(self, index: Union[int, slice]) -> Union[_T, "NDArray[_T]"]:
+    def __getitem__(self, index: Union[int, slice]) -> Union[_T, pa.list_]:
```
Comment (Lee-000) [inline graviti/utility/lazy.py:74]: Why the return type is `pa.list_`?

File: graviti/schema/extractors.py [hunk 3]
```diff
@@ -96,7 +96,7 @@ def extractor(data: Dict[str, Any]) -> Iterator[DataFrame]:
                 "ymax": [label["box2d"]["ymax"] for label in labels],
             }
             box2d.update(_extract_category_and_attribute(schema["items"], labels))
-            yield DataFrame(box2d)
+            yield box2d
 
     return extractor, "object"
```
Comment (Lee-000) [inline graviti/schema/extractors.py:101]: The return dtype is still `object`, is this acceptable?

File: graviti/schema/extractors.py [hunk 4]
```diff
@@ -20,10 +20,10 @@
 
 
 _PORTEX_TYPE_TO_NUMPY_DTYPE = {
```
Comment (Lee-000) [inline graviti/schema/extractors.py:22]: Should the name be updated?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

File: graviti/schema/extractors.py [hunk 5]
```diff
@@ -107,47 +95,45 @@ def _extract_vetices(
...
-    def extractor(data: Dict[str, Any]) -> Iterator[DataFrame]:
+def _get_pointlists(schema: Dict[str, Any], key: str) -> _Extractor[Dict[str, Any]]:
+    def extractor(data: Dict[str, Any]) -> Iterator[Dict[str, Any]]:
         for item in data["dataDetails"]:
             labels = item["label"][key.upper()]
             pointlists: Dict[str, Any] = {
-                "vertices": [
-                    DataFrame(_extract_vetices(label[key], ("x", "y"))) for label in labels
-                ]
+                "vertices": [list(_extract_vetices(label[key], ("x", "y"))) for label in labels]
```
Comment (Lee-000) [inline graviti/schema/extractors.py:103]: Is it ok to use `list` to convert the values of the dict here?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

**Review by Lee-000 — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**